components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -3045,6 +2888,425 @@ subroutine check_values(Qv,T,i,timestepcount,force_abort,source_ind)
+   real(rtype), intent(in) :: t 
+   real(rtype), intent(in) :: pres
+   real(rtype), intent(in) :: rhofaci 
+   real(rtype), intent(in) :: f1pr05
+   real(rtype), intent(in) :: f1pr14 
+   real(rtype), intent(in) :: xxlv
+   real(rtype), intent(in) :: xlf
+   real(rtype), intent(in) :: dv 
+   real(rtype), intent(in) :: kap 
+   real(rtype), intent(in) :: mu 
+   real(rtype), intent(in) :: sc
+   real(rtype), intent(in) :: qv 
+   real(rtype), intent(in) :: qc_incld 
+   real(rtype), intent(in) :: qitot_incld
+   real(rtype), intent(in) :: nitot_incld  
+   real(rtype), intent(in) :: qr_incld
+
+   logical, intent(inout) :: log_wetgrowth
+   real(rtype), intent(inout) :: qrcol 
+   real(rtype), intent(inout) :: qccol 
+   real(rtype), intent(inout) :: qwgrth
+   real(rtype), intent(inout) :: nrshdr 
+   real(rtype), intent(inout) :: qcshd 
+
+   real(rtype) :: qsat0, dum, dum1 
+
+
+   if (qitot_incld.ge.qsmall .and. qc_incld+qr_incld.ge.1.e-6_rtype .and. t.lt.zerodegc) then
+      qsat0  = 0.622_rtype*e0/(pres-e0)
+
+      qwgrth = ((f1pr05 + f1pr14*sc**thrd*(rhofaci*rho/mu)**0.5_rtype)*       &
+      2._rtype*pi*(rho*xxlv*dv*(qsat0-qv)-(t-zerodegc)*           &
+      kap)/(xlf+cpw*(t-zerodegc)))*nitot_incld
+
+      qwgrth = max(qwgrth,0._rtype)
+      dum    = max(0._rtype,(qccol+qrcol)-qwgrth)
+      if (dum.ge.1.e-10_rtype) then
+         nrshdr = nrshdr + dum*1.923e+6_rtype   ! 1/5.2e-7, 5.2e-7 is the mass of a 1 mm raindrop
+         if ((qccol+qrcol).ge.1.e-10_rtype) then
+            dum1  = 1._rtype/(qccol+qrcol)
+            qcshd = qcshd + dum*qccol*dum1
+            qccol = qccol - dum*qccol*dum1
+            qrcol = qrcol - dum*qrcol*dum1
+         endif
+         ! densify due to wet growth
+         log_wetgrowth = .true.
+      endif
+
+
+   end if 
+
+
+
+
+end subroutine wet_growth 
+
+
+subroutine ice_relaxation(rho,t,rhofaci,     &
+f1pr05,f1pr14,dv,mu,sc,qitot_incld,nitot_incld,    &
+epsi,epsi_tot)
+
+   !-----------------------------
+   ! calcualte total inverse ice relaxation timescale combined for all ice categories
+   ! note 'f1pr' values are normalized, so we need to multiply by N
+
+   implicit none 
+
+
+   real(rtype), intent(in) :: rho 
+   real(rtype), intent(in) :: t 
+   real(rtype), intent(in) :: rhofaci 
+   real(rtype), intent(in) :: f1pr05 
+   real(rtype), intent(in) :: f1pr14
+   real(rtype), intent(in) :: dv 
+   real(rtype), intent(in) :: mu 
+   real(rtype), intent(in) :: sc 
+   real(rtype), intent(in) :: qitot_incld 
+   real(rtype), intent(in) :: nitot_incld 
+
+   real(rtype), intent(out) :: epsi
+   real(rtype), intent(out) :: epsi_tot 
+
+
+
+   if (qitot_incld.ge.qsmall .and. t.lt.zerodegc) then
+      epsi = ((f1pr05+f1pr14*sc**thrd*(rhofaci*rho/mu)**0.5_rtype)*2._rtype*pi* &
+      rho*dv)*nitot_incld
+      epsi_tot   = epsi_tot + epsi
+   else
+      epsi = 0._rtype
+   endif 
+
+
+end subroutine ice_relaxation
+
+
+subroutine rime_density(t,rhofaci,    &
```

Comment:
just a general comment about all of the constants used in this routine.  Do any of these represent universal constants?  TBH I don't know the answer but I wanted to followup with, is there any reason we would consider changing some of these constants?  If so, should we make them locally defined constants that can be easily changed.

Reply:
@AaronDonahue That is a very good question, the only one that looks to me like a universal constant is 611, the others I am not sure of. I thought about making them local constants but that seemed outside of the present effort of breaking things into subroutines. That said, I am happy to do that if you think now is a good time.